assets/js/components/higherorder/withData.js

```diff
@@ -153,100 +176,141 @@ const withData = (
 				moduleRequiringSetup: '',
 			};
 
+			this.handleModuleDataReset = this.handleModuleDataReset.bind( this );
+			this.handleReturnedData = this.handleReturnedData.bind( this );
+			this.addDataRequests = this.addDataRequests.bind( this );
+			this.removeDataRequests = this.removeDataRequests.bind( this );
+		}
+
+		componentDidMount() {
 			addAction(
 				'googlesitekit.moduleDataReset',
 				'googlesitekit.moduleDataResetHandler',
-				() => {
-					this.setState( {
-						data: false,
-						errorMessage: false,
-						zeroData: false,
-					} );
+				this.handleModuleDataReset
+			);
+
+			this.addDataRequests();
+		}
+
+		componentWillUnmount() {
+			removeAction(
+				'googlesitekit.moduleDataReset',
+				'googlesitekit.moduleDataResetHandler',
+				this.handleModuleDataReset
+			);
+
+			this.removeDataRequests();
+		}
+
+		handleModuleDataReset() {
+			// When the global dateRange changes, it will trigger the googlesitekit.moduleDataReset action.
+			// When this happens, we need to re-hook the requests for the default date range to be applied correctly.
+			this.removeDataRequests();
+			this.addDataRequests();
+
+			this.setState( {
+				data: false,
+				errorMessage: false,
+				zeroData: false,
+			} );
+		}
+
+		addDataRequests() {
+			const { dateRange } = this.props;
+
+			Object.entries( dataRequestsByContext ).forEach(
+				( [ context, dataRequests ] ) => {
+					addFilter(
+						`googlesitekit.module${ context }DataRequest`,
+						`googlesitekit.withData.${ hashRequests( dataRequests ) }`,
+						( contextRequests ) => {
+							const requestsToAdd = [];
+							for ( const dataRequest of dataRequests ) {
+								const { type, identifier } = dataRequest || {};
+								// If a dataRequest's module requires setup, set it in the state.
+								// This will cause the setup incomplete component to be rendered in all cases.
+								if ( TYPE_MODULES === type && ! getModulesData()[ identifier ]?.setupComplete ) {
+									this.setState( { moduleRequiringSetup: identifier } );
+									continue;
+								}
+
+								// Apply default date range if not set.
+								const request = requestWithDateRange( dataRequest, dateRange );
```

Comment:
Here's where the default date range is applied now, where before it was done in `dataAPI.combinedGet`.